src/components/user.js

```diff
@@ -1,35 +1,81 @@
 import React from 'react';
 
+import UserInfo from '../stores/user-info';
+import UserAction from '../actions/user-action';
+
 import Address from './address.js';
 import Company from './company.js';
+import Info from './user-info';
+
 
 export default class User extends React.Component {
     constructor (props) {
         super(props);
 
+        const { info = {}, address = {}, company = {} } = this.props;
+
         this.state = {
+            currentUser: {
+                info,
+                address,
+                company
+            },
             showAddress: false,
-            showCompany: false
+            showCompany: false,
+            wantEdit:    false
         };
 
         this.handleClickAddress = this.handleClickAddress.bind(this);
         this.handleClickCompany = this.handleClickCompany.bind(this);
+        this.handleChange       = this.handleChange.bind(this);
+        this.handleClickEdit    = this.handleClickEdit.bind(this);
+        this.handleClickSubmit  = this.handleClickSubmit.bind(this);
     }
 
-    handleClickAddress () {
+    handleClickAddress (e) {
         this.setState({ showAddress: !this.state.showAddress });
+        e.preventDefault();
     }
 
-    handleClickCompany () {
+    handleClickCompany (e) {
         this.setState({ showCompany: !this.state.showCompany });
+        e.preventDefault();
+    }
+
+    handleClickEdit (e) {
+        const { currentUser } = this.state;
+        const newUser = new UserInfo(currentUser.info, currentUser.address, currentUser.company);
+
+        this.setState({ wantEdit: !this.state.wantEdit });
+
+        if (this.state.wantEdit) {
+            UserAction.updateUser(newUser);
+        }
+        e.preventDefault();
     }
 
+    handleChange (e, obj) {
+        const { currentUser } = this.state;
+
+        if (this.state.wantEdit || this.props.isNewUser) {
```

Comment:
You already use variable assigning from state object above. Let's add `wantEdit` there